python/pyarrow/array.pxi

```diff
@@ -3519,16 +3519,32 @@ class FixedShapeTensorArray(ExtensionArray):
     def to_numpy_ndarray(self):
         """
         Convert fixed shape tensor extension array to a numpy array (with dim+1).
+        """
+        cdef:
+            CFixedShapeTensorArray* ext_array = <CFixedShapeTensorArray*>(self.ap)
+            CResult[shared_ptr[CTensor]] ctensor
+        with nogil:
+            ctensor = ext_array.ToTensor()
+        return pyarrow_wrap_tensor(GetResultValue(ctensor)).to_numpy()
```

Comment:
I don't remember exactly what ToTensor does by default with regards to permutation (assuming it takes it into account), but so this expands the behaviour of `to_numpy_ndarray`, I think. In that case, can you expand the docstring and add some tests for this?

Reply:
`ToTensor` uses permutation to calculate strides and zero-copy creates `pyarrow.Tensor`.
`pyarrow.Tensor` can be zero-copy converted to a numpy array. I am not sure what that conversion does with strides, but we should check.
Updated the docstring.